domain/utxoindex/utxoindex.go

```diff
@@ -20,143 +17,143 @@ type UTXOIndex struct {
 }
 
 // New creates a new UTXO index
-func New(consensus externalapi.Consensus, database database.Database) *UTXOIndex {
-	store := newUTXOIndexStore(database)
-	return &UTXOIndex{
+func New(consensus externalapi.Consensus, database database.Database) (*UTXOIndex, error) {
```

Comment:
We should document that this should run *before* the flows. otherwise the virtual can be changed between `isSynced()` and registering